opencti-platform/opencti-graphql/src/database/middleware-loader.ts

```diff
@@ -292,26 +292,29 @@ export const buildRelationsFilter = <T extends BasicStoreCommon>(relationTypes:
 
 export const listRelations = async <T extends StoreProxyRelation>(context: AuthContext, user: AuthUser, type: string | Array<string>,
   args: RelationOptions<T> = {}): Promise<Array<T>> => {
-  const { indices = READ_RELATIONSHIPS_INDICES } = args;
+  const { indices } = args;
+  const computedIndices = computeQueryIndices(indices, type);
   const paginateArgs = buildRelationsFilter(type, args);
-  return elPaginate(context, user, indices, paginateArgs);
+  return elPaginate(context, user, computedIndices, paginateArgs);
 };
 
 export const listRelationsPaginated = async <T extends BasicStoreRelation>(context: AuthContext, user: AuthUser, type: string | Array<string>,
   args: RelationOptions<T> = {}): Promise<StoreRelationConnection<T>> => {
-  const { indices = READ_RELATIONSHIPS_INDICES, connectionFormat } = args;
+  const { indices, connectionFormat } = args;
+  const computedIndices = computeQueryIndices(indices, type);
   if (connectionFormat === false) {
     throw UnsupportedError('List connection require connectionFormat option to true');
   }
   const paginateArgs = buildRelationsFilter(type, args);
-  return elPaginate(context, user, indices, paginateArgs);
+  return elPaginate(context, user, computedIndices, paginateArgs);
 };
 
 export const listAllRelations = async <T extends StoreProxyRelation>(context: AuthContext, user: AuthUser, type: string | Array<string>,
   args: RelationOptions<T> = {}): Promise<Array<T>> => {
-  const { indices = READ_RELATIONSHIPS_INDICES } = args;
+  const { indices } = args;
```

Comment:
why remove here the default indices READ_RELATIONSHIPS_INDICES ? written like this, indices doesn't have a default value, and if type is not specified, computeQueryIndices(indices, type); will return READ_DATA_INDICES even though here we know that we want to read only relationships

Reply:
same question for all other methods where indices had a default value

Reply:
For me type is mandatory here. Do you see some usages without the type?

Reply:
you are right, type is always provided here